src/fparser/two/Fortran2003.py

```diff
@@ -7837,23 +7849,35 @@ class Block_Nonlabel_Do_Construct(BlockBase):  # pylint: disable=invalid-name
     subclass_names = []
     use_names = ["Nonlabel_Do_Stmt", "Execution_Part_Construct", "End_Do_Stmt"]
 
-    @staticmethod
-    def match(reader):
+    @classmethod
+    def match(cls, reader):
         """
         :param reader: instance of `FortranReaderBase` class
         :type reader: :py:class:`FortranReaderBase`
         :return: code block matching the nonlabeled "DO" construct
         :rtype: string
         """
         return BlockBase.match(
-            Nonlabel_Do_Stmt,
+            cls.nonlabel_do_stmt_cls(),
             [Execution_Part_Construct],
             End_Do_Stmt,
             reader,
             match_names=True,  # C821
             strict_match_names=True,  # C821
         )
 
+    @staticmethod
+    def nonlabel_do_stmt_cls():
+        """
+        :returns: Fortran2003 Nonlabel_Do_Stmt class.
+        :rtype: :py:class:`fparser.two.Fortran2003.Nonlabel_Do_Stmt`
+
+        """
+        # Import here to avoid circular imports
```

Comment:
pylint disable please.

Reply:
As above, not sure why this was added. Removed.